lib/Migration/RepairMailTheads.php

```diff
@@ -47,6 +48,16 @@ public function getName(): string {
 	}
 
 	public function run(IOutput $output): void {
+		/**
+		 * During the upgrade to v1.11.3 and later the old version of the
+		 * mapper is loaded before this new repair step is performed. Hence even after
+		 * the program code got replaced, the class doesn't have the new method. We
+		 */
+		if (!method_exists($this->mapper, 'resetInReplyTo')) {
+			$output->warning('New Mail code hasn\'t been loaded yet, skipping message ID repair. Please run `occ maintenance:repair` after the upgrade.');
+			return;
+		}
+
 		$count = $this->mapper->resetInReplyTo();
```

Comment:
@ChristophWurst @miaulalala Sorry, I am late here. But couldn't this method be located directly in the migration step, e.g. in the `postSchemaChange`? The method is only used in the repair step and you wouldn't need to run `occ maintenance:repair` afterwards manually.

Reply:
that would be possible as a workaround.

personally I would rather push for finally fixing repair steps than creating even more workaround everywhere